eng/common/TestResources/New-TestResources.ps1

```diff
@@ -92,8 +92,13 @@ param (
     [Parameter()]
     [switch] $SuppressVsoCommands = ($null -eq $env:SYSTEM_TEAMPROJECTID),
 
+    # Default behavior is to use -UserAuth
     [Parameter()]
-    [switch] $UserAuth,
+    [switch] $ServicePrincipalAuth,
+
+    [Parameter()]
+    [switch] $FederatedAuth,
```

Comment:
Is "FederatedAuth" the correct term here? It seems like that might be one form of logged in user but not all. Would "UseLoggedInCredentials" or something like that be better?

Reply:
`-UserAuth` already uses logged in credentials but it does other things like resolve a service principal ID from the logged in user and makes that the test app. 

Another approach could be to change `-UserAuth` to `-UseLoggedInCredential` and decide whether to resolve the service principal ID based on whether `-CI` is set.

Reply:
I think we could just remove `FederatedAuth` entirely. If `ServicePrincipalAuth` is not set, then we look for logged in credentials via azure powershell/azure cli login context (or try running Connect-AzAccount in the non-ci context). Then the federated auth parameter you have in the pipeline can just default to `-ServicePrincipalAuth:$${{ parameters.UseFederatedAuth }}`.

Reply:
So yes, CI and/or ServicePrincipalAuth parameters are what sets the mode/behavior.